src/Twilio/Security/RequestValidator.cs

```diff
@@ -129,17 +129,17 @@ private string RemovePort(UriBuilder uri)
             // UriBuilder.ToString() will not display the port 
             // if the Port property is set to -1
             uri.Port = -1;
-            return uri.ToString();
+            return uri.Uri.OriginalString;
         }
 
         private string AddPort(UriBuilder uri)
         {
             if (uri.Port != -1)
             {
-                return uri.ToString();
+                return uri.Uri.OriginalString;
             }
             uri.Port = uri.Scheme == "https" ? 443 : 80;
-            return uri.ToString();
+            return uri.Uri.OriginalString;
```

Comment:
This returns the original string passed to the constructor, so wouldn't this end up ignoring the addition of the port?

Reply:
I'm not sure, will double check the results of the TestValidateAddsPortHttp/s tests.

Reply:
Refactored this based on these test results using [this tool](https://docs.microsoft.com/en-us/dotnet/api/system.uri.originalstring?view=net-5.0 ):

```csharp
Uri uriAddress = new Uri("https://mycompany.com/myapp.php?foo=1&bar=2");
UriBuilder uriBuilder = new UriBuilder(uriAddress);
Console.WriteLine(uriAddress.ToString());
Console.WriteLine(uriBuilder.Uri.OriginalString);
// Resulting string: https://mycompany.com/myapp.php?foo=1&bar=2
Match m = Regex.Match(uriAddress.OriginalString, "^(([^:/?#]+):)?(//([^/?#]*))?([^?#]*)(\\?([^#]*))?(#(.*))?");
var hostName = m.Groups[4].ToString();
uriBuilder.Host = hostName.Split(':')[0];
uriBuilder.Port = uriBuilder.Scheme == "https" ? 443 : 80;
Console.WriteLine(uriBuilder.Uri.OriginalString);
// Resulting string: https://mycompany.com:443/myapp.php?foo=1&bar=2
uriBuilder.Host = hostName.Split(':')[0];
uriBuilder.Port = -1;
Console.WriteLine(uriBuilder.Uri.OriginalString);
// Resulting string: https://mycompany.com/myapp.php?foo=1&bar=2
 
uriAddress = new Uri("https://MyCompany.com/myapp.php?foo=1&bar=2");
uriBuilder = new UriBuilder(uriAddress);
Console.WriteLine(uriAddress.ToString());
Console.WriteLine(uriBuilder.Uri.OriginalString);
// Resulting string: https://mycompany.com:443/myapp.php?foo=1&bar=2
m = Regex.Match(uriAddress.OriginalString, "^(([^:/?#]+):)?(//([^/?#]*))?([^?#]*)(\\?([^#]*))?(#(.*))?");
hostName = m.Groups[4].ToString();
uriBuilder.Host = hostName.Split(':')[0];
uriBuilder.Port = uriBuilder.Scheme == "https" ? 443 : 80;
Console.WriteLine(uriBuilder.Uri.OriginalString);
// Resulting string: https://MyCompany.com:443/myapp.php?foo=1&bar=2
uriBuilder.Host = hostName.Split(':')[0];
uriBuilder.Port = -1;
Console.WriteLine(uriBuilder.Uri.OriginalString);
// Resulting string: https://MyCompany.com/myapp.php?foo=1&bar=2
```